src/Paprika/Chain/Blockchain.cs

```diff
@@ -1381,6 +1524,212 @@ public override string ToString() =>
             _ancestors.Select(ancestor => (ancestor.BlockNumber, ancestor.Hash));
     }
 
+    /// <summary>
+    /// BlockState implementation for snap sync - used only in RawState implementation
+    /// Handles boundary proof node data in a separate dictionary
+    /// </summary>
+    private class SyncBlockState : BlockState
+    {
+        private readonly PooledSpanDictionary _proofKeys;
+        private readonly HashSet<ulong> _proofHashSet;
```

Comment:
How big is it? Have you considered `BitFilter` here?

Reply:
This should not be big - max 2 x  height of the synced trie.